poppler: drop splash patch for 21.07.0

dylex opened the pull request:
`ENABLE_SPLASH` configuration has been removed entirely so patch is no longer necessary after #24931 (I think -- I'm not entirely sure why this was disabled to begin with).  Theoretically the `-DENABLE_SPLASH=OFF` could be conditionalized out as well, but it doesn't seem to be hurting anything.

alalazo commented on var/spack/repos/builtin/packages/poppler/package.py:
```diff
@@ -66,7 +66,7 @@ class Poppler(CMakePackage):
     # to be present.
     patch('poppler_page_splash.patch', when='@0.64.0:0.90.0 ^qt@5.0:')
-    patch('poppler_page_splash.0.90.1.patch', when='@0.90.1: ^qt@5.0:')
+    patch('poppler_page_splash.0.90.1.patch', when='@0.90.1:2 ^qt@5.0:')
```
```suggestion
    patch('poppler_page_splash.0.90.1.patch', when='@0.90.1 ^qt@5.0:')
```

alalazo reviewed (changes requested):
Can you check the suggestion for the patch, or explain better why `2` should be the upper bound of the condition?

dylex committed:
poppler: drop splash patch for 21.07.0

ENABLE_SPLASH configuration has been removed entirely after 21.06 so
patch is no longer necessary after #24931.  (Versions between 0.90.1 and
21.06 will likely still need a patch, and while it's not clear if this
patch is the right one, seems better to leave something in.)

dylex commented:
the `2` was mainly a place holder.  It looks like the relevant code was removed between 21.06 and 21.07 so I've changed this to `21.06`. It is possible this patch is not the correct one for such versions, but seems better to leave it in as an indication that something is needed, should such versions be added later.
